### PR TITLE
Singletons to hooks

### LINKED ISSUE #164: Convert AppSettings to hooks

### LINKED ISSUE #165: Convert Locale to hooks

### LINKED ISSUE #166: Convert Theme to hooks

### LINKED ISSUE #169: Convert ElementManager into hooks

### PR DESCRIPTION
**Pull request by mpeyper**
It's hard to seperate the singletons for this conversion as they use each other.  This PR currently contains refactoring for AppSettings, Locale and Theme, but will likely drag in ElementManager as well.

### CODE REVIEW

File: src/hooks/useLocalStorage.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    (value: T | ((currentValue: T) => T)) => {
+      try {
+        const valueToStore =
+          value instanceof Function ? value(storedValue) : value;
+        setStoredValue(valueToStore);
+        const serializedValue = JSON.stringify(valueToStore);
+        window.localStorage.setItem(key, serializedValue);
+        window.dispatchEvent(
+          new StorageEvent("storage", { key, newValue: serializedValue })
+        );
```
Comment (mpeyper) [inline src/hooks/useLocalStorage.ts:24]: This was weird.  Apparently storage events only fire in _other_ open tabs and windows, so to get the update to propagate to all the components int the _current_ window I had to dispatch the event manually (I got this from StackOverflow).

If we had a context provider that was hosting a single instance of the it, they propagation would be more automatic but _most_ components would then a wrapper in their tests.

Comment (JacobMGEvans) [reply src/hooks/useLocalStorage.ts:24]: I think this is a good workaround. Context still seems a bit like a heavy-handed approach with how these singletons were being implemented before. Also, I like the idea of not having the components codependent and integrated for such a simple state.

File: src/hooks/useElements.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+export function useElements() {
+  const { i18n } = useLocale();
+
+  const elements = useMemo(() => Object.values(elementsMap) as Element[], []);
+  const elementLookup = useMemo(
+    () =>
+      Object.fromEntries(elements.map((element) => [element.atomic, element])),
+    [elements]
+  );
```
Comment (HorusGoul) [inline src/hooks/useElements.ts:15]: This can be moved outside the hook as those won't change.

File: src/hooks/useTestSettings.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import * as React from "react";
+import { Element } from "@/Element";
+import { useSettings, Settings, ElementsSettings } from "./useSettings";
+import { useElements } from "./useElements";
+
+export function useTestSettings(
```
Comment (HorusGoul) [inline src/hooks/useTestSettings.ts:6]: Future tests won't use the same logic to store their settings, so we should mark this as legacy and deprecate its use. I'd call it `useLegacyTestSettings` and mark it as deprecated.

**Review by HorusGoul — CHANGES_REQUESTED** [review]

File: src/hooks/useTestSettings.ts (hunk 3 above)
Comment (JacobMGEvans) [reply src/hooks/useTestSettings.ts:6]: Sounds like a feature/enhancement, not really within the scope of refactor to Hooks

**Comment by JacobMGEvans** [comment]
@mpeyper I merging this as a working point for improvements and fixing that Cypress CI issue. I currently turned off the Cypress CI tests for now until its figured out.

File: src/hooks/useElements.ts (hunk 2 above)
Comment (JacobMGEvans) [reply src/hooks/useElements.ts:15]: @YPAzevedo can you make a PR for this if you have the time 😄

**Pull request merged**